crates/optimism/rpc/src/historical.rs

```diff
@@ -219,13 +167,146 @@ struct HistoricalRpcInner<P> {
     bedrock_block: BlockNumber,
 }
 
+impl<P> HistoricalRpcInner<P>
+where
+    P: BlockReaderIdExt + TransactionsProvider + Send + Sync + Clone,
+{
+    /// Checks if a request should be forwarded to the historical endpoint and returns
+    /// the response if it was forwarded.
+    async fn maybe_forward_request(&self, req: &Request<'_>) -> Option<MethodResponse> {
+        let should_forward = match req.method_name() {
+            "debug_traceTransaction" => self.should_forward_transaction(req),
+            method => self.should_forward_block_request(method, req),
+        };
+
+        if should_forward {
+            return self.forward_to_historical(req).await
+        }
+
+        None
+    }
+
+    /// Determines if a transaction request should be forwarded
+    fn should_forward_transaction(&self, req: &Request<'_>) -> bool {
+        parse_transaction_hash_from_params(&req.params())
+            .ok()
+            .map(|tx_hash| {
+                let tx_found =
+                    self.provider.transaction_by_hash_with_meta(tx_hash).ok().flatten().is_some();
+
```

Comment:
it might be the case that this chain imported the entire block history, in which case we would find the tx I believe.

since we're obtaining the additional metadata here as well we should also check, if we found the tx, that the meta.number is above/below the bedrock height, like we do with blocks